honeybadgermpc/hbavss.py

```diff
@@ -176,3 +181,152 @@ def _get_dealer_msg(self, value):
+                str(shared_key).encode("utf-8"),
+                encrypted_witnesses[self.my_id * (self.t + 1) + k])
+
+        # Don't know why this doesn't work
+        # # Decrypt
+        # for k in range(self.t+1):
+        #     shares[k], witnesses[k] = SymmetricCrypto.decrypt(
+        #         str(shared_key).encode("utf-8"), encrypted_witnesses[self.my_id][k])
+
+        # verify & send all ok
+        for i in range(self.t+1):
+            if self.poly_commit.verify_eval(
+                    commitments[i], self.my_id+1, shares[i], witnesses[i]):
+                multicast(HbAVSSMessageType.OK)
+            else:
+                # will be replaced by sending out IMPLICATE message later
+                # multicast(HbAVSSMessageType.IMPLICATE)
+                logging.error("PolyCommit verification failed.")
+                raise HoneyBadgerMPCError("PolyCommit verification failed.")
+
+        # Bracha-style agreement
+        oks_recvd = 0
+        readys_recvd = 0
+        # if 2t+1 OK or t+1 READY -> send all ready
+        while oks_recvd < 2 * self.t + 1 and readys_recvd < self.t + 1:
+            _, avss_msg = await recv()  # First value is the `sid`
+            if avss_msg == HbAVSSMessageType.OK:
+                oks_recvd += 1
+            elif avss_msg == HbAVSSMessageType.READY:
+                readys_recvd += 1
+
+        multicast(HbAVSSMessageType.READY)
+
+        # if 2t+1 ready -> output shares
+        while readys_recvd < 2 * self.t + 1:
+            _, avss_msg = await recv()  # First value is the `sid`
+            if avss_msg == HbAVSSMessageType.READY:
+                readys_recvd += 1
+
+        return shares
+
+    def _get_dealer_msg(self, value):
+        # Sample a random degree-(t,t) bivariate polynomial φ(·,·)
+        # such that each φ(0,k) = sk and φ(i,k) is Pi’s share of sk
+        phi = [None] * (self.t+1)
+        commitments = [None] * (self.t+1)
+        aux_poly = [None] * (self.t+1)
+        # for k ∈ [t+1]
+        #   Ck, auxk <- PolyCommit(SP,φ(·,k))
+        for k in range(self.t+1):
+            phi[k] = self.poly.random(self.t, value[k])
+            commitments[k], aux_poly[k] = self.poly_commit.commit(phi[k])
+
+        ephemeral_secret_key = self.field.random()
+        ephemeral_public_key = pow(self.g, ephemeral_secret_key)
+        # for each party Pi and each k ∈ [t+1]
+        #   1. w[i][k] <- CreateWitnesss(Ck,auxk,i)
+        #   2. z[i][k] <- EncPKi(φ(i,k), w[i][k])
+        z = [None] * self.n * (self.t + 1)
+        for i in range(self.n):
+            shared_key = pow(self.public_keys[i], ephemeral_secret_key)
+            for k in range(self.t + 1):
+                witness = self.poly_commit.create_witness(aux_poly[k], i+1)
+                z[i * (self.t+1) + k] = SymmetricCrypto.encrypt(
+                    str(shared_key).encode("utf-8"), (phi[k](i+1), witness))
+
+        # Don't know why this doesn't work
```

Comment:
What's this remaining commented out code, can it be removed?

Reply:
I'll remove them.

Reply:
Resolved